controllers/topology/internal/mergepatch/mergepatch_test.go

```diff
@@ -100,6 +100,34 @@ func TestNewHelper(t *testing.T) {
 			wantHasSpecChanges: false,
 			wantPatch:          []byte("{\"metadata\":{\"labels\":{\"foo\":\"bar\"}}}"),
 		},
+		{
```

Comment:
What should happen in the following case.

* .spec.machineTemplate.metadata is authoritative
* .spec.machineTemplate.metadata.labels is empty or does not exist in original
* .spec.machineTemplate.metadata.labels exist and has labels in modified

Should we delete the existing labels and remove the map?

Reply:
If spec.metadata is authoritative, the entire spec.metadata field is enforced, and thus in this case all the existing labels will be removed.

Reply:
Should we add a unit test for that in case we currently don't have one (sorry not up-to-date with the current state of the PR, will review again next week)

Reply:
Can we also add a test case to show that we're not overwriting finalizers in the modified object?

Reply:
It was covered by "Enforce nested value rebuilding struct if missing", but added even more coverage (note: I'm not using finalizers, but other field names)